### PR TITLE
Set strict mode on by default

### PR DESCRIPTION
**Pull request by seanconroy2021**
From the Slack discussion, we decided to update ec-cli to have strict mode 'true.' 
Jira : [Link here ](https://issues.redhat.com/browse/HACBS-2584)

### CODE REVIEW

**Comment by joejstuart** [comment]
LGTM. Just found `--strict` being used in a doc. https://github.com/seanconroy2021/ec-cli/blob/main/cmd/validate/image.go#L124

**Review by simonbaird — APPROVED** [review]

**Comment by simonbaird** [comment]
> LGTM. Just found `--strict` being used in a doc. https://github.com/seanconroy2021/ec-cli/blob/main/cmd/validate/image.go#L124

Yeah, that should probably be updated a little to indicate the change. Also this part:

```
404     cmd.Flags().BoolVarP(&data.strict, "strict", "s", data.strict,                                                                                
405         "return non-zero status on non-successful validation")   
```

Something like "return non-zero status on non-successful validation. Defaults to true. Use `--strict=false` to return a zero status code" or similar in both places.

**Review by simonbaird — CHANGES_REQUESTED** [review]
Removing approval to indicate I think a few doc wording changes would be a good idea.

**Comment by simonbaird** [comment]
> Something like "return non-zero status on non-successful validation. Defaults to true. Use `--strict=false` to return a zero status code" or similar in both places.

Or maybe just "return non-zero status on non-successful validation. Use `--strict=false` to return a zero status code even when there are policy violations", since I think the default value appears automatically.

**Comment by seanconroy2021** [comment]
> > Something like "return non-zero status on non-successful validation. Defaults to true. Use `--strict=false` to return a zero status code" or similar in both places.
> 
> Or maybe just "return non-zero status on non-successful validation. Use `--strict=false` to return a zero status code even when there are policy violations", since I think the default value appears automatically.

I will work on that this evening @simonbaird

**Comment by seanconroy2021** [comment]
> > Something like "return non-zero status on non-successful validation. Defaults to true. Use `--strict=false` to return a zero status code" or similar in both places.
> 
> Or maybe just "return non-zero status on non-successful validation. Use `--strict=false` to return a zero status code even when there are policy violations", since I think the default value appears automatically.

I will work on that this evening @simonbaird

File: cmd/validate/image.go [hunk 1]
```diff
@@ -402,12 +403,10 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
 		"[DEPRECATED] write output to a file. Use empty string for stdout, default behavior")
 
 	cmd.Flags().BoolVarP(&data.strict, "strict", "s", data.strict,
-		"return non-zero status on non-successful validation")
+		"Return non-zero status on non-successful validation. Defaults to true. Use --strict=false to return a zero status code.")
```
Comment (seanconroy2021) [inline cmd/validate/image.go:406]: @simonbaird Is this better

Comment (seanconroy2021) [reply cmd/validate/image.go:406]: comment from simonbarid: Yeah, that should probably be updated a little to indicate the change. Also this part:

404     cmd.Flags().BoolVarP(&data.strict, "strict", "s", data.strict,                                                                                
405         "return non-zero status on non-successful validation")   
Something like "return non-zero status on non-successful validation. Defaults to true. Use --strict=false to return a zero status code" or similar in both places.

File: cmd/validate/image.go [hunk 2]
```diff
@@ -402,12 +403,10 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
...
 
 	cmd.Flags().StringVar(&data.effectiveTime, "effective-time", policy.Now, hd.Doc(`
-		Run policy checks with the provided time. Useful for testing rules with
-		effective dates in the future. The value can be "now" (default) - for
-		current time, "attestation" - for time from the youngest attestation, or
+		Run policy checks with the preturn non-zstation" - for time from the youngest attestation, or
```
Comment (zregvart) [inline cmd/validate/image.go:409]: Should this be in this change? Seems off

**Review by zregvart — COMMENTED** [review]
Not sure about the change in the `effective-time` flag

File: cmd/validate/image.go (hunk 2 above)
Comment (seanconroy2021) [reply cmd/validate/image.go]: @zregvart don't know what happened shouldn't have been changed fixed it thank you.

**Review by zregvart — APPROVED** [review]
Perhaps a squash before merge?

**Comment by simonbaird** [comment]
Yeah, let's rebase and squash this into one tidy commit on top of main branch.

File: cmd/validate/image.go [hunk 3]
```diff
@@ -120,7 +121,7 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
 
 			Return a non-zero status code on validation failure:
 
-			  ec validate image --image registry/name:tag --strict
+			  ec validate image --image registry/name:tag
```
Comment (simonbaird) [inline cmd/validate/image.go:124]:
```suggestion
                        Return a zero status code even if there are validation failures:

			  ec validate image --image registry/name:tag --strict=false
```

File: features/validate_image.feature [hunk 4]
```diff
@@ -471,7 +471,7 @@ Feature: evaluate enterprise contract
       ]
     }
     """
-    When ec command is run with "validate image --snapshot acceptance/happy --policy acceptance/ec-policy --rekor-url ${REKOR} --public-key ${known_PUBLIC_KEY} --strict  --show-successes"
+    When ec command is run with "validate image --snapshot acceptance/happy --policy acceptance/ec-policy --rekor-url ${REKOR} --public-key ${known_PUBLIC_KEY}   --show-successes"
```
Comment (simonbaird) [inline features/validate_image.feature:474]: Nitpick: There are some extra spaces here and there, e.g. `before --show-successes` in this line and some of the others. It doesn't matter much, but you could consider could tidying them up.

**Comment by simonbaird** [comment]
Lgtm, but let's get rid of the merge commits, squash and rebase.

File: features/validate_image.feature (hunk 4 above)
Comment (seanconroy2021) [reply features/validate_image.feature]: Ok think i fixed all the spacing sorry for that.

**Comment by seanconroy2021** [comment]
@simonbaird this ready to be merged then

**Review by simonbaird — APPROVED** [review]

**Pull request merged**